examples/qt/widget.cpp

```diff
@@ -48,22 +48,25 @@ widget::widget(std::function<void(widget *, std::size_t)> callback, QWidget *par
     setLayout(main_layout);
 }
 
-void widget::add_label(std::size_t i, double t)
+void widget::threadsafe_add_label(std::size_t i, double t)
 {
-    std::lock_guard<hpx::lcos::local::spinlock> l(mutex);
+    // may be called from any thread, does not interact directly with GUI objects
     QString txt("Thread ");
     txt.append(QString::number(i))
        .append(" finished in ")
        .append(QString::number(t))
        .append(" seconds");
-    list->addItem(txt);
+
+    QGenericArgument arg("const QString&", &txt);
+    QMetaObject::invokeMethod(this, "add_label", arg);
```

Comment:
Doesn't this miss the `Qt::QueuedConnection` argument to make sure that it's properly dispatched?

Reply:
yes, Qt::ConnectionType is not specified, it thus should default to Qt::AutoConnection, which in turn should detect that you are in a different thread and use QueuedConnection.

It tested it with some "std::cout << get_thread_id() << std::endl;" calls that everything works as expected on my system.
But maybe adding the Qt::QueuedConnection argument would make the code more clear and readable.